### PR TITLE
SPUI-1373/1372 > Add history simplex

### PR DESCRIPTION
**Pull request by DenisMartych**

### CODE REVIEW

File: lib/app/shared/features/wallet/helper/operation_name.dart [hunk 1]
```diff
@@ -26,5 +26,7 @@ String operationName(OperationType type) {
       return 'Referral program bonus';
     case OperationType.unknown:
       return 'Unknown';
+    case OperationType.simplexBuy:
```
Comment (VovaMedvediev) [inline lib/app/shared/features/wallet/helper/operation_name.dart:29]: Swap OperationType.unknown and OperationType.simplexBuy, so unknown operation is last

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/components/common_transaction_details_block.dart [hunk 2]
```diff
@@ -29,11 +29,19 @@ class CommonTransactionDetailsBlock extends HookWidget {
 
     return Column(
       children: [
-        Text(
-          '${operationName(transactionListItem.operationType)} '
-          '${currency.description}',
-          style: sTextH5Style,
-        ),
+        if (transactionListItem.operationType != OperationType.simplexBuy)
```
Comment (VovaMedvediev) [inline lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/components/common_transaction_details_block.dart:32]: If statement is redundant. I think it is better to add varibale "header" ot something and place if-else logic in that variable.

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/buy_simplex_details.dart [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                  pressedIcon: const SCopyPressedIcon(),
+                ),
+              ],
+            ),
+          ),
+          const SpaceH14(),
+          TransactionDetailsItem(
+            text: 'With',
+            value: TransactionDetailsValueText(
+              text: '\$${transactionListItem.buyInfo!.sellAmount}',
```
Comment (VovaMedvediev) [inline lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/buy_simplex_details.dart:55]: Do we have assetSymbol in transactionListItem? So we don't hardcode "$" sign

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart [hunk 4]
```diff
@@ -100,7 +99,14 @@ class TransactionListItem extends HookWidget {
                     )}',
                     color: colors.grey2,
                   ),
-                ]
+                ],
+                if (transactionListItem.operationType ==
+                    OperationType.simplexBuy) ...[
```
Comment (VovaMedvediev) [inline lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart:104]: ...[] is redundant here. We have only one widget in if statament

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart [hunk 5]
```diff
@@ -137,6 +153,8 @@ class TransactionListItem extends HookWidget {
         return const SRewardPaymentIcon();
       case OperationType.unknown:
         return const SizedBox();
+      case OperationType.simplexBuy:
```
Comment (VovaMedvediev) [inline lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart:156]: replace unknown and simplexBuy

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/components/common_transaction_details_block.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/components/common_transaction_details_block.dart:32]: or extract if-else in private function that returns String

**Review by VovaMedvediev — CHANGES_REQUESTED** [review]

File: lib/app/shared/features/wallet/helper/operation_name.dart (hunk 1 above)
Comment (DenisMartych) [reply lib/app/shared/features/wallet/helper/operation_name.dart:29]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/components/common_transaction_details_block.dart (hunk 2 above)
Comment (DenisMartych) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/components/common_transaction_details_block.dart:32]: done, move to private function

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart (hunk 5 above)
Comment (DenisMartych) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart:154]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart (hunk 4 above)
Comment (DenisMartych) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart:104]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/buy_simplex_details.dart (hunk 3 above)
Comment (DenisMartych) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/buy_simplex_details.dart:55]: so far hardcode

**Pull request merged**